### PR TITLE
On Fuchsia with Vulkan on Acer, bypass Mozart and read input directly.

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

File: content_handler/direct_input.cc [hunk 1]
```diff
@@ -0,0 +1,230 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/content_handler/direct_input.h"
```
Comment (abarth) [inline content_handler/direct_input.cc]: blank line after primary header

File: content_handler/direct_input.cc [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+#include <dirent.h>
+#include <fcntl.h>
+#include <hid/acer12.h>
+#include <hid/usages.h>
+#include <magenta/device/device.h>
+#include <magenta/device/input.h>
+#include <sys/types.h>
+#include <sys/uio.h>
+#include <unistd.h>
```
Comment (abarth) [inline content_handler/direct_input.cc]: blank line after system headers

File: content_handler/direct_input.cc [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+#include <algorithm>
+#include <string>
+#include <utility>
+#include <vector>
```
Comment (abarth) [inline content_handler/direct_input.cc]: blank line after libc++ headers

File: content_handler/direct_input.cc [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+      input_event_.get(), DEVICE_SIGNAL_READABLE, MX_TIME_INFINITE, callback,
+      this);
+}
+
+void DirectInput::CancelWaitForReadAvailability() {
+  if (last_wait_ == 0) {
+    return;
+  }
+
+  fidl::GetDefaultAsyncWaiter()->CancelWait(last_wait_);
```
Comment (abarth) [inline content_handler/direct_input.cc]: I'd set last_wait_ to zero here to avoid double-free

**Comment by abarth** [comment]
LGTM

File: content_handler/direct_input.cc (hunk 3 above)
Comment (chinmaygarde) [inline content_handler/direct_input.cc]: Done. Fixed headers too.

**Pull request merged**